packaging/format-docs.py

```diff
@@ -0,0 +1,82 @@
+    for currentType in types:
+        if currentType["Namespace"] in typesByNamespace:
+            typesByNamespace[currentType["Namespace"]].append(currentType)
+        else:
+            typesByNamespace[currentType["Namespace"]] = [currentType]
+
+    explanation = ""
+    if collectionName == "TraitInfos":
+        explanation = "all traits with their properties and their default values plus developer commentary"
+    elif collectionName == "WeaponTypes":
+        explanation = "a template for weapon definitions as well as its contained types (warheads and projectiles) with default values and developer commentary"
+    elif collectionName == "SequenceTypes":
+        explanation = "all sequence types with their properties and their default values plus developer commentary"
+
+    print(f"This documentation is aimed at modders and has been automatically generated for version `{version}` of OpenRA. " +
+				"Please do not edit it directly, but instead add new `[Desc(\"String\")]` tags to the source code.\n")
+
+    print(f"Listed below are {explanation}:\n")
+
+    for namespace in typesByNamespace:
+        print(f'## {namespace}\n')
+
+        for currentType in typesByNamespace[namespace]:
+            print(f'### {currentType["Name"]}\n')
+
+            if currentType["Description"]:
+                print(f'#### {currentType["Description"]}\n')
+
+            if "InheritedTypes" in currentType and currentType["InheritedTypes"]:
+                print("Inherits from: " + ", ".join([format_type_name(x) for x in currentType["InheritedTypes"]]) + '.\n')
+
+            if "RequiresTraits" in currentType and currentType["RequiresTraits"]:
+                print("Requires trait(s): " + ", ".join([format_type_name(x) for x in currentType["RequiresTraits"]]) + '.\n')
+
+            if len(currentType["Properties"]) > 0:
+                hasAttributes = "OtherAttributes" in currentType["Properties"][0]
+                print(f'| Property | Default Value | Type |{" Attributes |" if hasAttributes else ""} Description |')
+                print(f'| -------- | ------------- | ---- |{" ---------- |" if hasAttributes else ""} ----------- |')
+
+                for prop in currentType["Properties"]:
+                    if "OtherAttributes" in prop:
+                        attributesList = [f'{x["Name"]}{"(" + str(x["Value"][0]) + ")" if x["Value"] else ""}' for x in prop["OtherAttributes"] if x["Name"]]
+                        if not attributesList:
+                            attributes = ''
+                        else:
+                            attributes = ", ".join(attributesList)
+
+                        print(f'| {prop["PropertyName"]} | {prop["DefaultValue"]} | {prop["UserFriendlyType"]} | {attributes} | {prop["Description"]} |')
+                    else:
+                        print(f'| {prop["PropertyName"]} | {prop["DefaultValue"]} | {prop["UserFriendlyType"]} | {prop["Description"]} |')
+
+            print('\n#\n')
```

Comment:
Why is this needed? Now after every entry a hash is placed. To match the old generation remove this line entirely

Reply:
I'm not trying to perfectly match the current format. I've always felt it makes it hard to tell where one entry starts and the other one ends, so I added a separator here.

Reply:
I doubt it would look good on the website

Reply:
It is unfortunate that I can't easily test how it would look there, but IMO it looks reasonable with GitHub's markdown rendering 🤷‍♂️